platform-tests/src/test/java/org/junit/platform/launcher/core/LauncherFactoryTests.java

```diff
@@ -48,6 +50,25 @@ void noopTestExecutionListenerIsLoadedViaServiceApi() {
 		assertThat(listener).isPresent();
 	}
 
+	@Test
+	void unusedTestExecutionListenerIsNotLoadedViaServiceApi() {
+		DefaultLauncher launcher = (DefaultLauncher) LauncherFactory.create();
+		List<TestExecutionListener> listeners = launcher.getTestExecutionListenerRegistry().getTestExecutionListeners();
+
+		Optional<TestExecutionListener> noop = listeners.stream().filter(
+			NoopTestExecutionListener.class::isInstance).findFirst();
+
+		Optional<TestExecutionListener> unused = listeners.stream().filter(
+			UnusedTestExecutionListener.class::isInstance).findFirst();
+
+		Optional<TestExecutionListener> anotherUnused = listeners.stream().filter(
+			AnotherUnusedTestExecutionListener.class::isInstance).findFirst();
+
+		assertThat(unused).isEmpty();
```

Comment:
```suggestion
		assertThat(listeners).filteredOn(UnusedTestExecutionListener.class::isInstance).isEmpty();
```

Reply:
This seems to update the list itself and can only be used once after which it'll filter the list leaving it blank

* Filter the iterable under test keeping only elements matching the given {@link Predicate}.

Reply:
That Javadoc is misleading, it does not modify the supplied list.